packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -327,20 +356,18 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     }
 
     /// @notice Updates gas config. Can only be called by the owner.
-    /// @param _overhead New overhead value.
-    /// @param _scalar   New scalar value.
-    function setGasConfig(uint256 _overhead, uint256 _scalar) external onlyOwner {
-        _setGasConfig(_overhead, _scalar);
+    /// @param _gasConfig New gas config.
+    function setGasConfig(GasConfig memory _gasConfig) external onlyOwner {
+        _setGasConfig(_gasConfig);
     }
 
     /// @notice Internal function for updating the gas config.
-    /// @param _overhead New overhead value.
-    /// @param _scalar   New scalar value.
-    function _setGasConfig(uint256 _overhead, uint256 _scalar) internal {
-        overhead = _overhead;
-        scalar = _scalar;
+    /// @param _gasConfig New gas config.
+    function _setGasConfig(GasConfig memory _gasConfig) internal {
+        overhead = _gasConfig.overhead;
+        scalar = _gasConfig.scalar;
 
-        bytes memory data = abi.encode(_overhead, _scalar);
+        bytes memory data = abi.encode(_gasConfig);
```

Comment:
I have a slight preference for not encoding the struct and instead encoding the struct properties so it looks the same or would like a unit test showing equivalence

Reply:
I unwrapped the properties in the implementation. 
There is also a test of the initializer which used the encoded struct, showing that they are the same.